src/target/riscv/riscv.c

```diff
@@ -849,27 +938,30 @@ static int remove_trigger(struct target *target, struct trigger *trigger)
 	if (riscv_enumerate_triggers(target) != ERROR_OK)
 		return ERROR_FAIL;
 
+	riscv_reg_t tselect;
+	int result = riscv_get_register(target, &tselect, GDB_REGNO_TSELECT);
+	if (result != ERROR_OK)
+		return result;
+
 	unsigned int i;
+	bool done = false;
 	for (i = 0; i < r->trigger_count; i++) {
```

Comment:
Just a detail: While at it, could you please move the `i` variable definition inside the loop:

```suggestion
	bool done = false;
	for (unsigned int i = 0; i < r->trigger_count; i++) {
```